### ISSUE TITLE
serialization with dask

### ISSUE THREAD

**Issue by simone-codeluppi** [soft-matter/pims#315]
Hi
I get an error of serialisation on dask when I am trying to split a subset of frames to different workers.
The error is similar to #178 
Have u ever tested pims in combination di dask?
Thanks a lot for the great work!

Simone

```

**Comment by nkeim** [comment]
I and others are also interested in this issue, including @jakirkham . The [dask-image](https://github.com/dask/dask-image) addresses this a bit, but it is separate from pims. Obviously the best solution would be to make a pims reader a full-fledged dask object that can be passed among workers as needed.

The good news is that (in my experience) the overhead of opening a pims reader is very low, much less than reading and processing an image. So if you want to just pass around an argument to `pims.open()`, `ImageSequence()`, or whatever, and create a new reader every time you need an image, you can. That's what I do.

**Comment by jakirkham** [comment]
Agreed. This would be great to solve. Was walking someone through what dask-image does on this front today.

IIRC there are some pickling issues that would need to be addressed. ( https://github.com/soft-matter/pims/issues/265 ) There's also some NumPy duck-typing behavior that would be good to have. ( https://github.com/soft-matter/pims/pull/254 )
